src/error/status.rs

```diff
@@ -5,120 +5,174 @@ use super::Result;
+
+    VOLUME_CORRUPTED        => 10, "An inconstancy was detected on the file \
+                                    system causing the operating to fail.",
+
+    VOLUME_FULL             => 11, "There is no more space on the file system.",
+
+    NO_MEDIA                => 12, "The device does not contain any medium to \
+                                    perform the operation.",
+
+    MEDIA_CHANGED           => 13, "The medium in the device has changed since \
+                                    the last access.",
+
+    NOT_FOUND               => 14, "The item was not found.",
+
+    ACCESS_DENIED           => 15, "Access was denied.",
+
+    NO_RESPONSE             => 16, "The server was not found or did not \
+                                    respond to the request.",
+
+    NO_MAPPING              => 17, "A mapping to a device does not exist.",
+
+    TIMEOUT                 => 18, "The timeout time expired.",
+
+    NOT_STARTED             => 19, "The protocol has not been started.",
+
+    ALREADY_STARTED         => 20, "The protocol has already been started.",
+
+    ABORTED                 => 21, "The operation was aborted.",
+
+    ICMP_ERROR              => 22, "An ICMP error occurred during the network \
+                                    operation.",
+
+    TFTP_ERROR              => 23, "A TFTP error occurred during the network \
+                                    operation.",
+
+    PROTOCOL_ERROR          => 24, "A protocol error occurred during the \
+                                    network operation.",
+
+    INCOMPATIBLE_VERSION    => 25, "The function encountered an internal \
+                                    version that was incompatible with a \
+                                    version requested by the caller.",
+
+    SECURITY_VIOLATION      => 26, "The function was not performed due to a \
+                                    security violation.",
+
+    CRC_ERROR               => 27, "A CRC error was detected.",
+
+    END_OF_MEDIA            => 28, "Beginning or end of media was reached",
+
+    END_OF_FILE             => 31, "The end of the file was reached.",
+
+    INVALID_LANGUAGE        => 32, "The language specified was invalid.",
+
+    COMPROMISED_DATA        => 33, "The security status of the data is unknown \
+                                    or compromised and the data must be \
+                                    updated or replaced to restore a valid \
+                                    security status.",
+
+    IP_ADDRESS_CONFLICT     => 34, "There is an address conflict address \
+                                    allocation",
+
+    HTTP_ERROR              => 35, "A HTTP error occurred during the network \
+                                    operation."
+}
+
+/// Macro to make implementation of warning status codes easier
+macro_rules! warning_codes {
+    ( $( $status:ident => $value:expr, $docstring:expr ),* ) => {
+        $(
+            #[doc = $docstring]
```

Comment:
Instead of using a separate string and the `#[doc]` attribute, we can make the macro forward attributes to the constant (see the [bitflags](
https://github.com/bitflags/bitflags/blob/cf2d7463676ae14036d751f8664b7a8e70a70e8b/src/lib.rs#L340) crate for an example):

```rs
$(#[$outer:meta])*
$( $status:ident => $value:expr, $docstring:expr ),*
```

and then:

```rs
$(#[$inner:ident $($args:tt)*])*
pub const ...
```

That way we can keep using the docstring format.

Reply:
Done :)